dev/src/Command/AddComponentCommand.php

```diff
@@ -184,21 +210,215 @@ protected function execute(InputInterface $input, OutputInterface $output)
+            $output->writeln($this->owlbotPostProcessor());
+        }
+
         $output->writeln('');
         $output->writeln('');
         $output->writeln('Success!');
 
         return 0;
     }
 
+    private function validateOptions(InputInterface $input, OutputInterface $output): void
+    {
+        $error = '<error>googleapis-gen-path and bazel-path options ' .
+            'are mutually exclusive. Will build library with bazel.</error>';
+        if ($input->getOption('bazel-path') && $input->getOption('googleapis-gen-path')) {
+            $output->writeln('');
+            $output->writeln($error);
+            $output->writeln('');
+        }
+    }
+
+    private function loadYamlConfigContent(NewComponent $new, string $protoDir): ?array
+    {
+        $yamlFilePath = sprintf('%s/%s%s%s.yaml',
+            $protoDir,
+            strtolower($new->shortName),
+            $new->version ? '_' : '',
+            $new->version ?? ''
+        );
+
+        try {
+            return Yaml::parse($this->loadProtoContent($yamlFilePath));
+        } catch (Exception $e) {
+            // Handle error gracefully.
+            return null;
+        }
+    }
+
     private function loadProtoContent(string $proto): string
     {
         if (file_exists($proto)) {
             return file_get_contents($proto);
         }
         $protoUrl = 'https://raw.githubusercontent.com/googleapis/googleapis/master/' . $proto;
-        $client = new Client();
-        $response = $client->get($protoUrl);
+        $response = $this->httpClient->get($protoUrl);
         return (string) $response->getBody();
     }
+
+    private function bazelQueryAndBuildLibrary(string $protoPath, string $googleApisDir): string
+    {
+        $command = ['bazel', '--version'];
+        $output = $this->runCommand($command);
+        // Extract the version number from the output
+        $match = preg_match('/bazel\s+(?P<version>\d+\.\d+\.\d+)/', $output, $matches);
+        if (!$match || $matches['version'] !== self::BAZEL_VERSION) {
+            throw new RuntimeException('Bazel 6.0.0 is not available');
+        }
+
+        $componentPath = str_replace(['(', ')'], '', $protoPath);
+        $command = [
+            'bazel',
+            'query',
+            'filter("-(php)$", kind("rule", //' . $componentPath .'/...:*))'
+        ];
+        $output = $this->runCommand($command, $googleApisDir);
+        $command = ['grep', '-v', '-E', ':(proto|grpc|gapic)-.*-php$'];
+        $output = $this->runCommand($command, $googleApisDir, $output);
+        // Remove new line character from the output, there will be only one component for one version.
+        return $this->bazelBuildLibrary(str_replace(["\n"], "", $output), $googleApisDir);
+    }
+
+    private function bazelBuildLibrary(string $component, string $googleApisDir): string
+    {
+        $command = ['bazel', 'build', $component];
+        return $this->runCommand($command, $googleApisDir, null, true);
```

Comment:
nit: why not combine these for simplicitly?
```suggestion
        $component = str_replace(["\n"], "", $output);
        $command = ['bazel', 'build', $component];
        return $this->runCommand($command, $googleApisDir, null, true);
```